b2b_ecommerce/models_test.py

```diff
@@ -52,3 +55,71 @@ def test_reference_number(settings):
         f"{REFERENCE_NUMBER_PREFIX}{cybersource_prefix}-{order.id}"
         == order.reference_number
     )
+
+
+def test_get_unexpired_coupon(order_with_coupon):
+    """get_unexpired_coupon should get a coupon matching the product id and coupon code, which is also valid"""
+    coupon = order_with_coupon.coupon
+    assert (
+        B2BCoupon.objects.get_unexpired_coupon(
+            coupon_code=coupon.coupon_code, product_id=coupon.product_id
+        )
+        == coupon
+    )
+
+
+@pytest.mark.parametrize(
+    "activation_date, expiration_date",
+    [
+        [None, None],
+        [timezone.now() - timedelta(days=1), timezone.now() + timedelta(days=1)],
+        [None, timezone.now() + timedelta(days=1)],
+        [timezone.now() - timedelta(days=1), None],
+    ],
+)
+def test_get_unexpired_coupon_null_dates(
+    order_with_coupon, activation_date, expiration_date
+):
+    """expiration or activation dates which are null should be treated as valid"""
+    coupon = order_with_coupon.coupon
+    coupon.activation_date = activation_date
+    coupon.expiration_date = expiration_date
+    coupon.save()
+    assert (
+        B2BCoupon.objects.get_unexpired_coupon(
+            coupon_code=coupon.coupon_code, product_id=coupon.product_id
+        )
+        == coupon
+    )
+
+
+@pytest.mark.parametrize(
+    "attr_name, attr_value",
+    [
+        ["enabled", False],
+        ["activation_date", timezone.now() + timedelta(days=1)],
+        ["expiration_date", timezone.now() - timedelta(days=1)],
+    ],
+)
+def test_get_unexpired_coupon_expired(order_with_coupon, attr_name, attr_value):
+    """get_unexpired_coupon should raise a B2BCoupon.DoesNotExist if the coupon is expired"""
```

Comment:
`get_unexpired_coupon should raise a B2BCoupon.DoesNotExist if there is no active, enabled, and unexpired coupon`